internal/build/tar.go

```diff
@@ -31,24 +31,18 @@ func archiveDf(ctx context.Context, tw *tar.Writer, df Dockerfile) error {
 	return nil
 }
 
-// archiveIfExists creates a tar archive of all local files in `paths` _if they exist_,
-// and returns a list of all PathMappings that point to nonexistent LocalPaths (we'll
-// use that later to add RM statements).
+// archiveIfExists creates a tar archive of all local files in `paths` _if they exist_.
 // NOTE: modifies tw in place.
-func archiveIfExists(ctx context.Context, tw *tar.Writer, paths []pathMapping) (dnePaths []pathMapping, err error) {
+func archiveIfExists(ctx context.Context, tw *tar.Writer, paths []pathMapping) error {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "daemon-archiveIfExists")
 	defer span.Finish()
 	for _, p := range paths {
-		dne, err := tarPath(ctx, tw, p.LocalPath, p.ContainerPath)
+		_, err := tarPath(ctx, tw, p.LocalPath, p.ContainerPath)
```

Comment:
update docstring and return values for tarPath if we're not using it to send back nonexistent paths?

Reply:
updated